be/src/olap/rowset/alpha_rowset_reader.cpp

```diff
@@ -201,24 +229,38 @@ OLAPStatus AlphaRowsetReader::_get_next_row_for_singleton_rowset(RowCursor** row
     return OLAP_SUCCESS;
 }
 
-OLAPStatus AlphaRowsetReader::_get_next_row_for_cumulative_rowset(RowCursor** row) {
-    size_t pos = 0;
-    if (_row_blocks[pos] == nullptr) {
-        return OLAP_ERR_DATA_EOF;
-    }
-    RowCursor* current_row = _row_cursors[pos];
-    OLAPStatus status = OLAP_SUCCESS;
-    if (!_row_blocks[pos]->has_remaining()) {
-        status = _get_next_block(pos, &_row_blocks[pos]);
-        if (status != OLAP_SUCCESS) {
-            LOG(WARNING) << "_get_next_block failed, status:" << status;
-            return status;
+OLAPStatus AlphaRowsetReader::_next_block_for_column_data(size_t pos, RowBlock** row_block) {
+    // get next block
```

Comment:
I think the read logic of first block can be merged in this function

Reply:
I will combine function call.
union_block and merge_block will be remained.